qiskit/transpiler/_transpiler.py

```diff
@@ -66,144 +65,53 @@ def transpile(circuits, backend, basis_gates=None, coupling_map=None, initial_la
-
+    circuits = parallel_map(_parallel_transpilation, circuits,
+                            task_args=(backend,),
+                            task_kwargs={'basis_gates': basis_gates,
+                                         'coupling_map': coupling_map,
+                                         'initial_layout': initial_layout,
+                                         'seed_mapper': seed_mapper,
+                                         'pass_manager': pass_manager})
     if return_form_is_single:
         return circuits[0]
     return circuits
 
 
-def _circuits_2_dags(circuits):
-    """Convert a list of circuits into a list of dags.
+def _parallel_transpilation(circuit, backend, basis_gates=None, coupling_map=None,
+                            initial_layout=None, seed_mapper=None,
+                            pass_manager=None):
+    """Perform transpilation of multiple circuits in parallel.
 
     Args:
-        circuits (list[QuantumCircuit]): circuits to convert
-
-    Returns:
-        list[DAGCircuit]: the dag representation of the circuits
-    """
-    dags = parallel_map(DAGCircuit.fromQuantumCircuit, circuits)
-    return dags
-
-
-def _dags_2_circuits(dags):
-    """Convert a list of dags into a list of circuits.
-
-    Args:
-        dags (list[DAGCircuit]): dags to convert
-
-    Returns:
-        list[QuantumCircuit]: the circuit representation of the dags
-    """
-    circuits = parallel_map(QuantumCircuit.fromDAGCircuit, dags)
-    return circuits
-
-
-def _dags_2_dags(dags, basis_gates='u1,u2,u3,cx,id', coupling_map=None,
-                 initial_layouts=None, seed_mapper=None, pass_manager=None):
-    """Transform multiple dags through a sequence of passes.
-
-    Args:
-        dags (list[DAGCircuit]): dag circuits to transform
-        basis_gates (str): a comma separated string for the target basis gates
-        coupling_map (list): A graph of coupling
-        initial_layouts (list[dict]): A mapping of qubit to qubit for each dag
-        seed_mapper (int): random seed_mapper for the swap mapper
-        pass_manager (PassManager): pass manager instance for the transpilation process
-            If None, a default set of passes are run.
-            Otherwise, the passes defined in it will run.
-            If contains no passes in it, no dag transformations occur.
+        circuit (QuantumCircuit): A circuit to transpile.
+        backend (BaseBackend): a backend to compile for
+        basis_gates (str): comma-separated basis gate set to compile to
+        coupling_map (list): coupling map (perhaps custom) to target in mapping
+        initial_layout (list): initial layout of qubits in mapping
+        seed_mapper (int): random seed for the swap_mapper
+        pass_manager (PassManager): a pass_manager for the transpiler stage
 
     Returns:
-        list[DAGCircuit]: the dag circuits after going through transpilation
+        list[QuantumCircuit]: Transpiled circuit(s)
```

Comment:
This function returns just a single QuantumCircuit

Reply:
yes